library/core/src/result.rs

```diff
@@ -224,6 +224,21 @@
 //! [`Err(E)`]: Err
 //! [io::Error]: ../../std/io/struct.Error.html "io::Error"
 //!
+//! # Representation
+//!
+//! In some cases, [`Result<T, E>`] will gain the same size, alignment, and ABI
+//! guarantees as [`Option<T>`] has. One of either the `T` or `E` type must be a
```

Comment:
I found this a bit confusing the first couple of times I read it, because I thought the `T` in `Result<T,E>` and in `Option<T>` were the same `T`.

Maybe change to `Option<U>` (or other, then explain the non-empty one is `U`?

Reply:
that's a good point but i just turned off the PC so I'll do that tomorrow